opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -171,30 +172,30 @@ def __init__(
         self._metric_readers = metric_readers
 
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader._register_measurement_consumer(self)
+            metric_reader._set_collect_callback(
+                self._measurement_consumer.collect
+            )
 
+        self._shutdown_once = Once()
         self._shutdown = False
 
     def force_flush(self) -> bool:
 
         # FIXME implement a timeout
 
-        metric_reader_result = True
-
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader_result = (
-                metric_reader_result and metric_reader.force_flush()
-            )
-
-        if not metric_reader_result:
-            _logger.warning("Unable to force flush all metric readers")
-
-        return metric_reader_result
+            metric_reader.collect()
+        return True
 
     def shutdown(self):
         # FIXME implement a timeout
 
-        if self._shutdown:
+        def _shutdown():
+            self._shutdown = True
```

Comment:
@lonewolf3739 there is also #2239

There is a slight chance of L223 racing a call to shutdown. The fix would be allow passing a lock into the `Once` class (similar to Condition) and using the same lock on L223.

However, I feel like it's also not a big deal for that race to exist (compared with executing shutdown more than once) and might be ok to just close out that issue. Wdyt?

Reply:
Sounds good to me. I think it is very uncommon to run into that possibility but even then I don't see much issue.

Reply:
@aabmass I see you approved, please resolve this conversation if you are ok with @lonewolf3739 answer so that we can merge this PR :+1: Thanks!